pkg/metricscollector/v1beta1/file-metricscollector/file-metricscollector_test.go

```diff
@@ -93,28 +95,124 @@ func TestCollectObservationLog(t *testing.T) {
+							Name:  "accuracy",
+							Value: "0.8078",
+						},
+					},
+					{
+						TimeStamp: "2024-03-04T17:55:08Z",
+						Metric: &v1beta1.Metric{
+							Name:  "loss",
+							Value: "0.5183",
+						},
+					},
+					{
+						TimeStamp: "2024-03-04T17:55:08Z",
+						Metric: &v1beta1.Metric{
+							Name:  "accuracy",
+							Value: "0.6752",
+						},
+					},
+					{
+						TimeStamp: "2024-03-04T17:55:08Z",
+						Metric: &v1beta1.Metric{
+							Name:  "loss",
+							Value: "0.3634",
+						},
+					},
+					{
+						TimeStamp: "2024-03-04T17:55:08Z",
+						Metric: &v1beta1.Metric{
+							Name:  "accuracy",
+							Value: "100",
+						},
+					},
+					{
+						TimeStamp: "2024-03-04T17:55:08Z",
+						Metric: &v1beta1.Metric{
+							Name:  "accuracy",
+							Value: "888.333",
+						},
+					},
+					{
+						TimeStamp: "2024-03-04T17:55:08Z",
+						Metric: &v1beta1.Metric{
+							Name:  "accuracy",
+							Value: "-0.4759",
+						},
+					},
+					{
+						TimeStamp: time.Time{}.UTC().Format(time.RFC3339),
+						Metric: &v1beta1.Metric{
+							Name:  "loss",
+							Value: "0.8671",
+						},
+					},
+				},
+			},
 		},
-		{
-			description: "Invalid file format",
-			filePath:    filepath.Join(testJsonDataPath, "good.json"),
-			fileFormat:  "invalid",
-			err:         true,
+		"Invalid case for logs in TEXT format": {
+			filePath:   filepath.Join(testTextDataPath, "invalid-value.log"),
+			filters:    []string{"{metricName: ([\\w|-]+), metricValue: ((-?\\d+)(\\.\\d+)?)}"},
+			metrics:    []string{"accuracy", "loss"},
+			fileFormat: commonv1beta1.TextFormat,
+			expected: &v1beta1.ObservationLog{
+				MetricLogs: []*v1beta1.MetricLog{
+					{
+						TimeStamp: time.Time{}.UTC().Format(time.RFC3339),
+						Metric: &v1beta1.Metric{
+							Name:  "accuracy",
+							Value: consts.UnavailableMetricValue,
+						},
+					},
+				},
+			},
 		},
-		{
-			description: "Invalid formatted file for logs in JSON format",
-			filePath:    filepath.Join(testJsonDataPath, "invalid-format.json"),
-			fileFormat:  commonv1beta1.JsonFormat,
-			err:         true,
+		"Invalid file name": {
+			filePath: "invalid",
+			err:      true,
 		},
-		{
-			description: "Invalid timestamp for logs in JSON format",
-			filePath:    filepath.Join(testJsonDataPath, "invalid-timestamp.json"),
-			fileFormat:  commonv1beta1.JsonFormat,
-			metrics:     []string{"acc", "loss"},
+		"Invalid file format": {
+			filePath:   filepath.Join(testTextDataPath, "good.log"),
+			fileFormat: "invalid",
+			err:        true,
+		},
+		"Invalid formatted file for logs in JSON format": {
+			filePath:   filepath.Join(testJsonDataPath, "invalid-format.json"),
+			fileFormat: commonv1beta1.JsonFormat,
+			err:        true,
+		},
+		"Invalid formatted file for logs in TEXT format": {
+			filePath:   filepath.Join(testTextDataPath, "invalid-format.log"),
+			filters:    []string{"{metricName: ([\\w|-]+), metricValue: ((-?\\d+)(\\.\\d+)?)}"},
```

Comment:
@Electronic-Waste @tenzen-y Why if file has incorrect format we return error for JSON format, but for TEXT we make metrics Unavailable ?

Reply:
Because invalid format errors can be detected in JSON format. But in TEXT, we can't distinguish the invalid format error from missing metrics.

Reply:
That makes sense!